sensors/triggers/params.go

```diff
@@ -203,52 +225,51 @@ func ResolveParamValue(src *v1alpha1.TriggerParameterSource, events map[string]*
 	case src.Value != nil:
 		// Use the default value set by the user in case the event is missing
 		resultValue = *src.Value
-		return &resultValue, nil
+		return &resultValue, stringType, nil
 	default:
 		// The parameter doesn't have a default value and is referencing a dependency that is
 		// missing in the received events. This is not an error and may happen with || conditions.
-		return nil, nil
+		return nil, errorType, nil
 	}
-
 	// If the event payload parsing failed
 	if err != nil {
 		// Fall back to the default value in case it exists
 		if src.Value != nil {
 			fmt.Printf("failed to parse the event payload, using default value. err: %+v\n", err)
 			resultValue = *src.Value
-			return &resultValue, nil
+			return &resultValue, stringType, nil
 		}
-
 		// Otherwise, return the error
-		return nil, err
+		return nil, errorType, err
 	}
-
 	// Get the value corresponding to specified key or template within event payload
 	if eventPayload != nil {
 		if tmplt != "" {
 			resultValue, err = getValueWithTemplate(eventPayload, tmplt)
 			if err == nil {
-				return &resultValue, nil
+				return &resultValue, errorType, nil
 			}
 			fmt.Printf("failed to execute the src event template, falling back to key or value. err: %+v\n", err)
 		}
 		if key != "" {
-			resultValue, err = getValueByKey(eventPayload, key)
+			tmp, typ, err := getValueByKey(eventPayload, key)
+			// For block injection support
+			resultValue = tmp
 			if err == nil {
-				return &resultValue, nil
+				return &resultValue, typ, nil
 			}
 			fmt.Printf("failed to get value by key: %+v\n", err)
 		}
 		// In case neither key nor template resolving was successful, fall back to the default value if exists
 		if src.Value != nil {
 			resultValue = *src.Value
-			return &resultValue, nil
+			return &resultValue, typ, nil
```

Comment:
Does the `typ` returned here have a value?

Reply:
Right... I will return errorType, because we fail the function

Reply:
It falls back to the default value, does not fail the function.

Reply:
What do you suggest to return here?